### DISCUSSION TITLE
[C128] Segment 'LOWCODE' doesn't seem to go into Common (none banking) RAM

### DISCUSSION THREAD

**Discussion by RetrofanC** [cc65/cc65#1593] [Q&A] [answered]
Hi, 

As functional I have things are working now the next step is further automating things. 
Besides, trying to keep things as much as 'standard cc65' as possible. 

To that end I have the following question; 

In the c128.cfg there are the following segements defined; 

STARTUP:     load = MAIN,     type = ro;
LOWCODE:  load = MAIN,     type = ro,  optional = yes;
ONCE:          load = MAIN,     type = ro,  optional = yes;

Now the suggestion is that 'LOWCODE' should be in COMMON RAM and not Banked.

Also as per the description of, per example,  the irq.s file;

; To solve this, we place
; the following code into a special segment called "LOWCODE" which will be
; placed just above the startup code, so it goes into a RAM area that is
; not banked.

.segment        "LOWCODE"

When I look into the extended MAP file I find the locations for LOWCODE here; 

__LOWCODE_LOAD__          001C68 RLA    
__LOWCODE_RUN__           001C68 RLA    
__LOWCODE_SIZE__          000008

Based on my understand in the CC65 starts with C128 in the mode with just 1k of COMMON RAM. 

Furthermore, I don't see any code where this code from $1c68 is brought back (memcpy()) under the $0400 range to fit into the 1k COMMON RAM. 

For my project I want to make preferable one (or two) MEMORY sections and SEGMENT for all code which has to go into the COMMON RAM. So like to understand the workings of LOWCODE better to avoid conflicts. 

Thanks in advance,

**Comment by RetrofanC** [comment]
Hi, 

 I also checked some of the other .cfg files like nes.cfg, plus4.cfg, cx16-banking.cfg and all are using the 'LOWCODE' segment.
But for all of them it is just using the MAIN/RAM0 memory profile. 

For none of them any sign they use a 'special' part of the that memory. 

The quest continues...

**Comment by oliverschmidt** [comment, marked as answer]
For most linker configs coming with cc65, the LOWCODE segment doesn't go into a special memory area - you are completely right about that. The idea behind this setup is this:

Banking mechanisms on 6502 systems tend to work on high(er) addresses. So content tends to be more safe from being banked out when it is placed at lower addresses. The LOWCODE segment is placed before the CODE segment - and therefore at lower addresses (thus the name). This makes its content tend to be more safe from being banked out. No more, no less.

**Reply by RetrofanC** [reply]
Thanks Oliver, most helpful answer indeed! 

So from my side let me see what I can do to optimize the situation for the specific c128 environment and circle back.
